src/json.h

```diff
@@ -703,6 +706,71 @@ class JsonObject
 
         // useful debug info
         std::string line_number(); // for occasional use only
+
+        template <typename T>
```

Comment:
The part where you're adding non-json logic to the json module, it looks like this code was added to generic_factory.h, but there's still a copy of it here.